File: src/main/java/codeTop/ms/LeetCode300.java
package codeTop.ms;

/**
 * @author jingxinwu
 * @date 2022-02-17 8:46 PM
 */
public class LeetCode300 {

    public int lengthOfLIS(int[] nums) {

        if (nums == null || nums.length == 0) {
            return 0;
        }
        int[] dp = new int[nums.length];
        dp[0] = 1;
        //fix res
        int res = 1;
        for (int i = 1; i < nums.length; i++) {
            int temp = 0;
            for (int j = 0; j < i; j++) {
                if (nums[i] > nums[j]) {
                    temp = Math.max(temp, dp[j]);
                }
            }
            dp[i] = temp + 1;
            res = Math.max(res, dp[i]);
        }
        return res;
    }

}

File: src/main/go/leetcode/lists/hot100/LeetCode300.go
package hot100

func lengthOfLIS(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	dp := make([]int, len(nums))
	res := 1
	for i := 0; i < len(nums); i++ {
		temp := 0
		for j := 0; j < i; j++ {
			if nums[i] > nums[j] {
				temp = Max(temp, dp[j])
			}
		}
		dp[i] = temp + 1
		res = Max(res, dp[i])
	}
	return res
}
